src/analysis/Environments/Environments.ts

```diff
@@ -136,7 +136,9 @@ export const Environments = (props: EnvironmentsProps): ReactNode => {
   const [shouldFilterByCreator, setShouldFilterByCreator] = useState(true);
 
   const refreshData = withLoading(async () => {
-    await refreshWorkspaces();
+    if (runtimes) {
```

Comment:
This has the behavior of not refreshing on initial load, but refreshing on subsequent effects. So it fixes the double-call causing this bug.